src/ASPNetCoreMastersTodoList.Api/Filters/EnsureItemExistsAttribute.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Services;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ASPNetCoreMastersTodoList.Api.Filters
+{
+    public class EnsureItemExistsFilter : IActionFilter
+    {
+        public readonly IItemService itemService;
+
+        public EnsureItemExistsFilter(IItemService itemService)
+        {
+            this.itemService = itemService ?? throw new ArgumentNullException(nameof(itemService));
+        }
+
+        public void OnActionExecuted(ActionExecutedContext context)
+        {
+        }
+
+        public void OnActionExecuting(ActionExecutingContext context)
+        {
+            var itemId = (int)context.ActionArguments["id"];
```

Comment:
please handle scenario when the filter is not applicable because we need it as controller scope

Reply:
Scenario handled